src/memory.rs

```diff
@@ -114,3 +121,51 @@ pub fn alloc_cons<'a, 'b, A>(
         OCaml::new(cr, ocaml_cons)
     }
 }
+
+const BOX_OPS_DYN_DROP: custom_operations = custom_operations {
+    identifier: "_rust_box_dyn_drop\0".as_ptr() as *const i8,
+    finalize: Some(drop_box_dyn),
+    compare: None,
+    hash: None,
+    serialize: None,
+    deserialize: None,
+    compare_ext: None,
+    fixed_length: ptr::null(),
+};
+
+extern "C" fn drop_box_dyn(oval: RawOCaml) {
```

Comment:
@g2p why is `Any` needed? is it because you have a single "destructor" functions for every type?

Reply:
&dyn Any is a fat pointer, and that allows checking that we downcast to the correct type.
Downcasting is used for dropping and for the borrow implementation (in value.rs).
And, yes, it's necessary because this is with a single custom_operations vtable.

For type safety, I could go further and check that the custom block is of the right type by checking its custom_ops pointer. If we add specialized custom types later, these can rely on the custom_ops pointer check without the need for a Rust fat pointer.

Those kind of extra checks would be good if they were triggered from OCamlClosure::handle_call_results at least; currently the call* methods are unsound since all types are picked by the caller (think Obj.magic).  I think I can make a separate proposal for this.